qhana_plugin_runner/api/tasks_api.py

```diff
@@ -85,19 +100,40 @@ def get(self, task_id: str):
             task_data.task_id is not None
         ), "If this is None then get_task_by_id is faulty."  # assertion for type checker
 
+        progress = None
+        steps = None
+        if task_data.multi_step:
+            progress = {
+                "value": task_data.progress_value,
+                "start": task_data.progress_start,
+                "target": task_data.progress_target,
+                "unit": task_data.progress_unit,
+            }
+            steps: List[StepMetadata] = []
+            step: Step
+            for step in task_data.steps:
+                steps.append(
+                    StepMetadata(
+                        href=step.href,
+                        uiHref=step.ui_href,
+                        stepId=step.step_id,
+                        cleared=step.cleared,
+                    )
+                )
+
         if not task_data.is_finished:
-            task_result = AsyncResult(task_id, app=CELERY)
+            task_result = AsyncResult(task_data.task_id, app=CELERY)
```

Comment:
I removed the case 
```
return TaskData(
    progress=progress,
    steps=steps,
    status=task_data.status,
    task_log=task_data.task_log,
)
```
task_log should enable the display of adequate error messages in case of some error during a (multi-step) execution.

Reply:
Yes, task log should now always be included when it is not empty. Event when the task is still pending.